lib-injection/sitecustomize.py

```diff
@@ -159,15 +170,21 @@ def package_is_compatible(package_name, package_version):
     return installed_version.version >= supported_version_spec.version
 
 
+def executable_is_compatible():
+    return sys.argv[0] not in EXECUTABLES_DENY_LIST and sys.argv[1] not in EXECUTABLES_DENY_LIST
```

Comment:
It might be useful to drop a comment here--at least, I'm not sure why `sys.argv[1]` is being checked.  AFAIK, the convention for Python (and some other interpreted languages) is to provide its own view that differs from the libc argv.

For instance, if I do
```
python3 -I foo.py`
```
then the system argv will be `[".../python3", "-I", "foo.py"]` but the python `sys.argv` will just be `["foo.py"]`

I think this matters because certain parts of this stack will use the libc argv (as is passed through `execve()`), but our side of the fence lives within the interpreter, and folks who don't immediately realize every time that they aren't the same might cargo-cult the wrong ideas (in particular, the libc-based approach needs to do a much more complete analysis of the arg strings to arrive to the same conclusion).

I believe this statement is still at least partially valid if I'm wrong about `sys.argv` :)

I have a tool for  examples of this [here](https://github.com/sanchda/systems_experiments/tree/main/argv0) if you'd like to validate the behavior for yourself.